src/language/lexer.rs

```diff
@@ -2,26 +2,26 @@
 use std::str::CharIndices;
 use std::iter::Peekable;
 
-#[derive(Debug, PartialEq, Eq)]
+#[derive(Debug, PartialEq)]
 pub enum Token<'a> {
-  Eof,
-  Bang,
-  Dollar,
-  LeftParen,
-  RightParen,
-  Spread,
-  Colon,
-  Equals,
-  At,
-  LeftBracket,
-  RightBracket,
-  LeftBrace,
-  RightBrace,
-  Pipe,
-  Name(&'a str),
-  Int(&'a str),
-  Float(&'a str),
-  String(&'a str),
+  Eof(usize, usize),
+  Bang(usize, usize),
+  Dollar(usize, usize),
+  LeftParen(usize, usize),
+  RightParen(usize, usize),
+  Spread(usize, usize),
+  Colon(usize, usize),
+  Equals(usize, usize),
+  At(usize, usize),
+  LeftBracket(usize, usize),
+  RightBracket(usize, usize),
+  LeftBrace(usize, usize),
+  RightBrace(usize, usize),
+  Pipe(usize, usize),
+  Name(usize, usize, &'a str),
+  Int(usize, usize, &'a str),
+  Float(usize, usize, &'a str),
```

Comment:
As a note, the spec models tokens differently where `Bang`, `Dollar`, etc. are modeled by a `Punctuator` type. So the types might instead look like this:

```rust
pub enum Token<'a> {
  Eof(…),
  Punctuator(usize, Punctuator),
  IntValue(…),
  FloatValue(…),
  StringValue(…)
}

pub enum Punctuator {
  Bang,
  Dollar,
  LeftParen,
  …
}
```

I think this might make code cleaner later when trying to `match` against tokens to parse into grammar types. Where there is only one `usize` because we can guarantee there will only be a single character (might be wrong on the one `usize` there, I have no experience writing lexer/parsers 😊).

At the very least I think you should change `Int`, `Float`, and `String` to `IntValue`, `FloatValue`, and `StringValue` per the [specification](http://facebook.github.io/graphql/#sec-Appendix-Grammar-Summary.Lexical-Tokens).